packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/Relations/Relations.tsx

```diff
@@ -971,11 +981,34 @@ const ListItem = ({ data, index, style }: ListItemProps) => {
     name,
     relations,
     targetModel,
+    mainField,
   } = data;
+  const { currentDocumentMeta } = useDocumentContext('RelationsField');
 
   const { formatMessage } = useIntl();
 
-  const { href, id, label, status, documentId, apiData, locale } = relations[index];
+  const {
+    href,
+    id,
+    label: originalLabel,
+    status: originalStatus,
+    documentId,
+    apiData,
+    locale,
+  } = relations[index];
+
+  const collectionType = getCollectionType(href)!;
+  const { document } = useDocument(
+    {
+      collectionType,
+      model: targetModel,
+      documentId: documentId ?? apiData?.documentId,
+      params: currentDocumentMeta.params,
+    },
+    { skip: !apiData?.temporary }
+  );
+  const label = document != null ? getRelationLabel(document, mainField) : originalLabel;
+  const status = document !== null ? document?.status : originalStatus;
```

Comment:
It would make more sense to have that ternary depend on the temporary thing imo, so it matches the skip param above